### DISCUSSION TITLE
Can you use this with the Puppeteer stealth plugin?

### DISCUSSION THREAD

**Discussion by solkad** [spatie/browsershot#685] [General]
Since this uses Puppeteer, I was wondering if there was any way to use browsershot with the Puppeteer stealth plugin.

https://www.npmjs.com/package/puppeteer-extra-plugin-stealth

Is this possible? If so, how would one do it?

**Comment by kriiv** [comment]
I'm also curious about this - would be extremely useful.

**Comment by rony98** [comment]
I ended up getting puppeteer extra plugin working with Browsershot. I don't necessarily recommend doing this as changing the vendor folder is never recommended but I did the following:

npm install puppeteer-extra-plugin-stealth
npm install puppeteer puppeteer-extra

To install puppeter extra and the stealth plugin.

Went to vendor/spatie/browsershot/bin to edit the browser.js file

and within this file I changed line 60, "const puppet = (pup || require('puppeteer'));" to the following 4 lines:

    const puppet = (pup || require('puppeteer-extra'));
    const StealthPlugin = require('puppeteer-extra-plugin-stealth');
    puppet.use(StealthPlugin());

    const {executablePath} = require('puppeteer');
    
The new executablePath is required because for whatever reason the old one being used wasn't working with newer version of puppeteer-core (I didn't look too much into it, found the solution on stackoverflow). With that I also had to change the current line 90 (originally line 86) to:

executablePath: executablePath(),

To use the new executable path and it worked. 

Once again I don't necessarily recommend this. I think a PR should be made into Browershot to make the change properly, without hacking it, and so the vendor folder isn't messed with directly.

**Reply by ao-jhelmich** [reply]
You might want to check out the following pull request #399 

This does the same thing you are describing with the tools delivered by the library

**Reply by rony98** [reply]
Will do, thank you.

**Comment by patrykszady** [comment]
Hi, I'm having trouble implementing this. Tried the solution above as well as https://github.com/spatie/browsershot/pull/399. I'm not sure where to place the libs/browsershot.js file and where to point setBinPath to.
